File: 0053-maximum-subarray/0053-maximum-subarray.cpp
class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        vector<int>dp(nums.size());
        dp[0] = nums[0];
        int res = nums[0];
        for(int i = 1; i < nums.size(); i++) {
            dp[i] = max(nums[i], nums[i] + dp[i - 1]);
            res = max(dp[i], res);
        }
        return res;
    }
};

File: 0053-maximum-subarray/0053-maximum-subarray.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        
        max_sum = float("-inf")
        cSum = 0

        for num in nums:
            cSum = max(cSum + num, num)
            max_sum = max(max_sum, cSum)
        
        return max_sum